util/topgen/merge.py

```diff
@@ -240,6 +240,29 @@ def elaborate_instance(instance, block: IpBlock):
             raise ValueError(f'generate_dif contains invalid value {instance["generate_dif"]}')
         instance['generate_dif'] = converted_value
 
+    # An instance can either have a 'racl_mapping' or 'racl_mappings' but can't have both.
+    # 'racl_mapping' is used when the device only has a single register interface and
+    # 'racl_mappings' there are more.
+    racl_mapping = instance.get('racl_mapping')
+    racl_mappings = instance.get('racl_mappings')
+
+    if racl_mapping and racl_mappings:
+        raise ValueError("Cannot specify both 'racl_mapping' and 'racl_mappings'")
+
```

Comment:
I'd suggest converting `racl_mapping` to a value for `racl_mappings` here, which will make the code below a bit easier to read.